OAuth login returns 500: missing SessionMiddleware and init_oauth never called

## Bug

OAuth login (Google, GitHub, Apple) crashes with a 500 Internal Server Error immediately on use.

**Two root causes introduced in #272 (feat: invitation + password + OAuth auth):**

1. **Missing `SessionMiddleware`** — The OAuth flow writes to `request.session` (`web.py:762-763`) and reads from it in the callback (`web.py:801-802`), but `SessionMiddleware` is never added to the FastAPI app. Starlette raises `AssertionError: SessionMiddleware must be installed to access request.session`.

2. **`init_oauth()` never called** — `oauth.py` defines `init_oauth(app)` which calls `_configure_providers()` and `oauth.init_app(app)`, but it is never invoked from `create_app()` or anywhere else. OAuth providers are never registered, so `create_client(provider)` always returns `None` → 404.

**Secondary issues in the OAuth callback (`web.py:766-859`):**

3. No error handling around `client.authorize_access_token(request)` — any OAuth provider error (denied permission, expired state, network failure) is an unhandled exception → 500.
4. No validation that `provider_uid` is non-empty before querying the database.

## Steps to reproduce

1. Configure any OAuth provider env vars (e.g. `OAUTH_GOOGLE_CLIENT_ID`)
2. Navigate to `/login` and click an OAuth provider button
3. 500 Internal Server Error

## Expected behavior

OAuth login redirects to provider, completes callback, and creates a session.

## Affected version

Commit 91232e0 (`feat: invitation + password + OAuth auth (#268) (#272)`)

--- a/.env.example
+++ b/.env.example
@@ -31,6 +31,7 @@ AVATAR_DIR=data/avatars
 # SMTP_FROM=helmlog@example.com
 # Authentication (#268 — invitation + flexible auth)
 # AUTH_DISABLED=true          # bypass auth entirely — local/LAN dev only
+# SESSION_SECRET=             # secret key for session cookie signing (OAuth state); auto-generated if unset
 # How long session cookies stay valid (days)
 AUTH_SESSION_TTL_DAYS=90
 # OAuth providers (optional — only providers with configured vars are enabled)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -42,6 +42,7 @@ dependencies = [
     "types-shapely>=2.1.0.20250917",
     "authlib>=1.3",
     "argon2-cffi>=23.1",
+    "itsdangerous>=2.2.0",
 ]
 
 [project.scripts]

--- a/src/helmlog/oauth.py
+++ b/src/helmlog/oauth.py
@@ -8,7 +8,6 @@ from __future__ import annotations
 
 import os
 import time
-from typing import Any
 
 from authlib.integrations.starlette_client import OAuth
 from loguru import logger
@@ -89,7 +88,6 @@ def enabled_providers() -> list[str]:
     return providers
 
 
-def init_oauth(app: Any) -> None:  # noqa: ANN401
-    """Initialize OAuth with the FastAPI/Starlette app."""
+def init_oauth() -> None:
+    """Register OAuth providers that have env vars configured."""
     _configure_providers()
-    oauth.init_app(app)

--- a/src/helmlog/web.py
+++ b/src/helmlog/web.py
@@ -310,6 +310,25 @@ def create_app(
     app = FastAPI(title="HelmLog", docs_url=None, redoc_url=None)
     app.state.limiter = limiter
     app.add_exception_handler(RateLimitExceeded, _rate_limit_exceeded_handler)  # type: ignore[arg-type]
+
+    # SessionMiddleware is required by Authlib's OAuth integration (request.session)
+    from starlette.middleware.sessions import SessionMiddleware
+
+    session_secret = os.getenv("SESSION_SECRET", "")
+    if not session_secret:
+        from helmlog.auth import generate_token
+
+        session_secret = generate_token()
+        logger.warning(
+            "SESSION_SECRET not set — generated an ephemeral key (OAuth state will not survive restarts)"
+        )
+    app.add_middleware(SessionMiddleware, secret_key=session_secret)
+
+    # Initialize OAuth providers
+    from helmlog.oauth import init_oauth
+
+    init_oauth()
+
     app.state.storage = storage
     _audio_session_id: int | None = None
     _debrief_audio_session_id: int | None = None
@@ -771,29 +790,51 @@ def create_app(
         if client is None:
             raise HTTPException(status_code=404, detail=f"Unknown OAuth provider: {provider}")
 
-        tok = await client.authorize_access_token(request)
+        try:
+            tok = await client.authorize_access_token(request)
+        except Exception as exc:
+            logger.warning("OAuth token exchange failed for {}: {}", provider, exc)
+            return HTMLResponse(
+                "<h1>OAuth login failed.</h1><p>Could not complete sign-in. Please try again.</p>",
+                status_code=400,
+            )
+
         if provider == "google":
             user_info = tok.get("userinfo", {})
             provider_email = user_info.get("email", "")
             provider_uid = user_info.get("sub", "")
             provider_name = user_info.get("name")
         elif provider == "github":
-            resp = await client.get("user")
-            user_info = resp.json()
-            provider_email = user_info.get("email", "")
-            provider_uid = str(user_info.get("id", ""))
-            provider_name = user_info.get("name")
-            if not provider_email:
-                email_resp = await client.get("user/emails")
-                for e in email_resp.json():
-                    if e.get("primary"):
-                        provider_email = e["email"]
-                        break
+            try:
+                resp = await client.get("user")
+                user_info = resp.json()
+                provider_email = user_info.get("email", "")
+                provider_uid = str(user_info.get("id", ""))
+                provider_name = user_info.get("name")
+                if not provider_email:
+                    email_resp = await client.get("user/emails")
+                    for e in email_resp.json():
+                        if e.get("primary"):
+                            provider_email = e["email"]
+                            break
+            except Exception as exc:
+                logger.warning("GitHub API call failed: {}", exc)
+                return HTMLResponse(
+                    "<h1>OAuth login failed.</h1><p>Could not retrieve GitHub profile. Please try again.</p>",
+                    status_code=400,
+                )
         else:  # apple
             user_info = tok.get("userinfo", {})
             provider_email = user_info.get("email", "")
             provider_uid = user_info.get("sub", "")
             provider_name = user_info.get("name")
+
+        if not provider_uid:
+            logger.warning("OAuth provider {} returned empty user ID", provider)
+            return HTMLResponse(
+                "<h1>OAuth login failed.</h1><p>Provider did not return a user identifier.</p>",
+                status_code=400,
+            )
 
         # Look up existing credential
         cred = await storage.get_credential_by_provider_uid(provider, provider_uid)

